lib/ansible/modules/cloud/vmware/vmware_get_info.py

```diff
@@ -0,0 +1,162 @@
+        object_type=dict(type='str',
+                         choices=['datacenter', 'cluster',
+                                  'datastore', 'folder',
+                                  'host', 'network',
+                                  'resource_pool', 'virtual_machine',
+                                  'content_library', 'local_library',
+                                  'subscribed_library', 'content_type',
+                                  'tag', 'category'],
+                         default='datacenter'),
+        filters=dict(type='list', default=[]),
+    )
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=True)
+
+    conn = Connection(module._socket_path)
+
+    rest_result = {}
+    object_type = module.params['object_type']
+    object_type_url_map = dict(
+        datacenter=('/vcenter/datacenter', ['folders', 'datacenters', 'names']),
+        cluster=('/vcenter/cluster', ['folders', 'datacenters', 'names', 'clusters']),
+        datastore=('/vcenter/datastore', ['folders', 'datacenters', 'names', 'datastores', 'types']),
+        folder=('/vcenter/folder', ['folders', 'datacenters', 'names', 'parent_folders', 'type']),
+        host=('/vcenter/host', ['folders', 'datacenters', 'names', 'hosts', 'clusters', 'connection_states']),
+        network=('/vcenter/network', ['folders', 'datacenters', 'names', 'types', 'networks']),
+        resource_pool=('/vcenter/resource-pool', ['datacenters', 'names', 'hosts', 'parent_resource_pools', 'clusters', 'resource_pools']),
+        virtual_machine=('/vcenter/vm', ['folders', 'datacenters', 'names', 'hosts', 'power_states', 'clusters', 'resource_pools', 'vms']),
+        content_library=('/com/vmware/content/library', []),
+        local_library=('/com/vmware/content/local-library', []),
+        subscribed_library=('/com/vmware/content/subscribed-library', []),
+        content_type=('/com/vmware/content/type', []),
+        category=('/com/vmware/cis/tagging/category', []),
+        tag=('/com/vmware/cis/tagging/tag', []),
+    )
+
+    object_url = '/rest' + object_type_url_map[object_type][0]
+
+    filters = module.params.get('filters')
+    if filters:
+        first = True
+        for filter in filters:
+            for key in list(filter.keys()):
+                lower_key = key.lower()
+                if lower_key not in object_type_url_map[object_type][1]:
+                    # Check if filter is valid for current object type or not
+                    continue
+                if not first:
+                    object_url += '&'
+                else:
+                    object_url += '?'
+                    first = False
+                # Escape characters
+                if '/' in filter[key]:
+                    filter[key].replace('/', '%2F')
+                object_url += 'filter.%s=%s' % (lower_key, filter[key])
+
+    return_code, response = conn.send_request(object_url, {}, method='GET')
```

Comment:
If the logic of generation object_url is common across VMware module it can be moved either in VMware modules utils or within VMware httpapi plugin as a function (eg: get_info()) depending on how generic the functionality is across VMware products.

Reply:
@ganeshrn Thanks for reviewing this PR. Since this is the first implementation and we yet to know the use cases of this. I will keep this in mind and will change in the future PR.